### DISCUSSION TITLE
images Disappearing

### DISCUSSION THREAD

**Discussion by strawbs1989** [community/community#124583] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

General

### Body

hi sorry for the messages, but i do not know why my images 
are disappearing on my website, they've before(this morning and yesterday) 
but why arent they displaying now? 

wondering if anyone could take a look [https://github.com/strawbs1989/CoolvibesR](url)
all my image directory's are correct img files such as png jpgs are correct 

but don't know why there disappearing?

**Comment by davevad93** [comment]
Hi @strawbs1989 ,

I've took a look at the images in your repo, the ones with `jpg` extension are fine but none of the `png` ones can be displayed which I suspect it's due to file corruption. Try to convert all your `png` images to `jpg`. I don't know why, but I've seen this behavior before when pushing `png` images to GitHub it must be something about binary values being corrupted.

**Comment by strawbs1989** [comment]
Binary values being corrupted? That's very weird, 😕 ok I'll try an convert
the remaining PNG to jpg n let you know if that works. But in the meantime
time can tech fix the behaviour of binary values?

**Reply by davevad93** [reply]
I really don't know why but look, in the meantime I've found this about `png` files being corrupted after a `git push`:

1. [.gitattributes corrupt png and videos files after push](https://github.com/bencodezen/vue-enterprise-boilerplate/issues/136)
2. [Images corrupt after git push](https://stackoverflow.com/questions/19411981/images-corrupt-after-git-push)

**Reply by strawbs1989** [reply]
@davevad93  all my images are jpg in my index.html, but really DO NOT know why there not showing up?

**Reply by davevad93** [reply]
Ok, that's strange, I thought the problem was with the `png` ones because they cannot be displayed in your repo, I've take a look now at your website and I've seen that only 5 images are displayed, I've checked your `index.html` too and everything looks fine, no errors, I don't know what might be causing the issue honestly..

**Reply by strawbs1989** [reply]
@davevad93 hmm neither do i to be honest, mind you earlier on yesterday uk time i did create a codespace and commit from there 
then deleted the codespace as it were to hard for me lol would an extra .github folder with a workflow folder inside 
affect the site or not????? if not i don't know what could be causing the images not to show!!

but if you have time to hunt through my repo and find the cause.... let me know and i'll try and fix it :)

**Reply by davevad93** [reply]
Hey @strawbs1989 , if I have time tomorrow I will take a look at it and let you know if I'm able to find the cause, now I'm going to rest, I live in Spain and here it's 3 in the morning eheh. I don't know where you are from so good day or good night.

**Reply by strawbs1989** [reply]
@davevad93 thank you let me know the cause i'm in hastings east sussex uk 2:30am here ehehe 
g'night

**Reply by davevad93** [reply]
Hey @strawbs1989 , did you finally solve the problem? I had another look at it but honestly I couldn't find the cause.

**Comment by RafaelJohn9** [comment]
> wondering if anyone could take a look [https://github.com/strawbs1989/CoolvibesR](https://github.com/orgs/community/discussions/url)

Your hyperlink is broken by the way the display link itself is okay.

I'll also check out to see the issue you are having above on images display.

**Reply by strawbs1989** [reply]
when i look at my index.html on local in a browser all my images show up? 
so somewhere in repo is stopping all my images from showing on github also 
i can get onto it [https://github.com/strawbs1989/CoolvibesR](url)

**Reply by RafaelJohn9** [reply]
hmm...weird ...when I click on the hyperlink its giving me a 404 error
instead of " {https://github.com/strawbs1989/CoolvibesR}(https://github.com/orgs/community/discussions/url)" try this {https://github.com/strawbs1989/CoolvibesR}([https://github.com/strawbs1989/CoolvibesR])

had to change the square brackets with braces in order for you to understand what I mean

**Comment by strawbs1989** [comment]
Think I have but not sure, it's something to do with .GitHub workflow
static.yml that's failing apparently!!!

I don't understand .yml files lol

**Comment by strawbs1989** [comment]
That's weird, like I say I can view the repo
Don't know why you can't?

**Reply by RafaelJohn9** [reply]
its a minor thing don't mind lemme continue checking up on the issue

**Comment by strawbs1989** [comment]
I've got this URL https://github.com/strawbs1989/CoolvibesR

But don't know if you can get onto my repo there, anyway yea let me the
issue please!!!!

**Reply by RafaelJohn9** [reply]
some of your images are corrupted
![image](https://github.com/community/community/assets/125447154/4be2a06f-22e4-44c0-8be0-11e0292065e3)
I can't even download them from GitHub without them being corrupt,

**Reply by RafaelJohn9** [reply]
try this  Here’s how you can ensure Git treats PNG files correctly:

    Add a .gitattributes file:
    Create a .gitattributes file in the root of your repository if you don’t already have one.

    Specify binary handling for PNG files:
    Add the following line to your .gitattributes file:
 ```
*.png binary
```

This tells Git to treat all .png files as binary, preventing any text-based transformations that could corrupt the file.

Re-add PNG files:
If PNG files have already been committed and corrupted, you will need to re-add them. First, remove the corrupted files and then re-add them. Here’s how you can do it:

``` bash

git rm --cached *.png
git add *.png
git commit -m "Re-adding PNG files as binary"
```
This process removes the files from the index (staging area) but not from the working directory, then re-adds them correctly as binary files.

**Reply by strawbs1989** [reply]
ok ta i'll try that i do have a .gitattributes but i have this in the file 

`* text eol=lf
*.{cmd,[cC][mM][dD]} text=eol=crlf
*.{bat,[bB][aA][tT]} text=eol=crlf`

ive tried what you told me and it worked!
but three images are still disappearing

**Reply by RafaelJohn9** [reply]
its good we are making progress 
lemme continue checking
have you deployed yet the latest version

**Reply by strawbs1989** [reply]
i haver yes! @RafaelJohn9

**Reply by RafaelJohn9** [reply]
Add the following line to your .gitattributes file:
```
*.jpg binary
*.jpeg binary
```
it should work after that

**Reply by RafaelJohn9** [reply]
@strawbs1989  lemme know if it doesn't

**Comment by strawbs1989** [comment]
I've already added .jpg .jpeg binarys
To .gitatributes file still nothing

**Reply by RafaelJohn9** [reply]
hmmmm...thats weird, can you confirm that it works locally

**Comment by strawbs1989** [comment]
Yes works locally

**Reply by RafaelJohn9** [reply, marked as answer]
at this point am seeing two solutions:
 - Try removing the images from the repo and re-adding them
 - Replace  the images with other images

**Comment by strawbs1989** [comment]
I'll try

**Reply by RafaelJohn9** [reply]
cool, lemme know

**Reply by RafaelJohn9** [reply]
if you wouldn't mind, can you mark one of my responses as the answer? would mean a lot.

**Comment by strawbs1989** [comment]
I can now confirm that all my images
Are now officially back 😁 check

https://coolvibes-reloaded.com

**Reply by RafaelJohn9** [reply]
👌🥳🥳congrats bruv

**Comment by RafaelJohn9** [comment]
>if you wouldn't mind, can you mark one of my responses as the answer? would mean a lot.

about this 😁😁😁

**Comment by RafaelJohn9** [comment]
@strawbs1989 thanks, if you have any question or problem am happy to help you can always reach out

**Comment by strawbs1989** [comment]
Nice to know there's someone out there I can reach out to
